File: baekjoon_all/02000+/boj_2884.cpp
// Solve 2021-12-18
// Update 2023-10-09

#include <iostream>
using namespace std;

int main() {
    int hour, minute;
    cin >> hour >> minute;

    minute -= 45;

    if (minute < 0) {
        minute += 60;
        hour--;
    }

    if (hour < 0) {
        hour += 24;
    }

    cout << hour << ' ' << minute << '\n';

    return 0;
}

File: baekjoon_all/02000+/boj_2884.java
// Solve 2023-11-17

// 백준에 제출할 때는 class 이름을 Main으로 설정해야 함.

import java.util.Scanner;

public class boj_2884 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int hour = sc.nextInt();
        int minute = sc.nextInt();

        minute -= 45;

        if (minute < 0) {
            minute += 60;
            hour--;
        }

        if (hour < 0) {
            hour += 24;
        }

        System.out.println(hour + " " + minute);

        sc.close();
    }
}
